akka-http2-support/src/main/scala/akka/http/impl/engine/http2/StreamIdBidiFlow.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.impl.engine.http2
+
+import akka.NotUsed
+
+import akka.http.scaladsl.Http2
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
+import akka.stream.scaladsl.BidiFlow
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+import akka.stream.{ Attributes, BidiShape, Inlet, Outlet }
+
+import scala.collection.mutable
+
+class StreamIdBidiFlow extends GraphStage[BidiShape[HttpRequest, HttpRequest, HttpResponse, HttpResponse]] {
+  val in = Inlet[HttpRequest]("StreamIdBidi.in")
+  val out = Outlet[HttpResponse]("StreamIdBidi.out")
+  val toWrapped = Outlet[HttpRequest]("StreamIdBidi.toWrapped")
+  val fromWrapped = Inlet[HttpResponse]("StreamIdBidi.fromWrapped")
+
+  override def initialAttributes = Attributes.name("StreamIdBidi")
+
+  val shape = BidiShape(in, toWrapped, fromWrapped, out)
+
+  override def toString = "One2OneBidi"
+
+  override def createLogic(effectiveAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
+    private val idsInFlight = mutable.ArrayDeque[Option[Int]]()
```

Comment:
This has the same problem as the One2OneBidi implementation in that it ever only pulls a single element. It needs a parallelism parameter that governs how many requests are pulled and then buffers them until they are fetched from downstream.

In the best case, this flow should be part of `FlowWithContext` (and maybe we already have it?) as a general way to add context propagation to ordered one-to-one Flows.